### PR TITLE
Fixed the OriginalSender encoding in nodeTransactions.go

### PR DESCRIPTION
**Pull request by iulianpascalau**
- fixed original sender encoding in nodeTransactions.go, `prepareUnsignedTx` function

### CODE REVIEW

**Review by sasurobert — DISMISSED** [review]

File: node/nodeTransactions_test.go [hunk 1]
```diff
@@ -464,3 +466,52 @@ func setupGetMiniblockMetadataByTxHash(historyRepo *testscommon.HistoryRepositor
...
+		Tx:             scr2,
+		Nonce:          3,
+		Type:           string(transaction.TxTypeUnsigned),
+		Value:          "4",
+		Receiver:       "erd1qcrqvpsxqcrqvpsxqcrqvpsxqcrqvpsxqcrqvpsxqcrqvpsxqcrqwkh39e",
+		Sender:         "erd1q5zs2pg9q5zs2pg9q5zs2pg9q5zs2pg9q5zs2pg9q5zs2pg9q5zsrqsks3",
+		OriginalSender: "erd1qurswpc8qurswpc8qurswpc8qurswpc8qurswpc8qurswpc8qurstywtnm",
+	}
+	assert.Equal(t, scrResult2, expectedScr2)
+
```
Comment (SebastianMarian) [inline node/nodeTransactions_test.go:516]: remove empty line :)

**Review by SebastianMarian — DISMISSED** [review]

File: node/nodeTransactions_test.go [hunk 2]
```diff
@@ -464,3 +466,52 @@ func setupGetMiniblockMetadataByTxHash(historyRepo *testscommon.HistoryRepositor
...
+	scrResult1, err := n.prepareUnsignedTx(scr1)
+	assert.Nil(t, err)
+	expectedScr1 := &transaction.ApiTransactionResult{
+		Tx:       scr1,
+		Nonce:    1,
+		Type:     string(transaction.TxTypeUnsigned),
+		Value:    "2",
+		Receiver: "erd1qyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqsl6e0p7",
+		Sender:   "erd1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq6gq4hu",
```
Comment (bogdan-rosianu) [inline node/nodeTransactions_test.go:492]: maybe check explicity that the `OriginalSender` is nil/empty

File: node/nodeTransactions.go [hunk 3]
```diff
@@ -339,7 +339,11 @@ func (n *Node) prepareUnsignedTx(tx *smartContractResult.SmartContractResult) (*
 		CodeMetadata:            tx.GetCodeMetadata(),
 		PreviousTransactionHash: hex.EncodeToString(tx.GetPrevTxHash()),
 		OriginalTransactionHash: hex.EncodeToString(tx.GetOriginalTxHash()),
-		OriginalSender:          n.addressPubkeyConverter.Encode(tx.GetOriginalSender()),
 		ReturnMessage:           string(tx.GetReturnMessage()),
-	}, nil
+	}
+	if len(tx.GetOriginalSender()) == n.addressPubkeyConverter.Len() {
+		txResult.OriginalSender = n.addressPubkeyConverter.Encode(tx.GetOriginalSender())
```
Comment (bogdan-rosianu) [inline node/nodeTransactions.go:345]: what is the difference? encoding an invalid address would have returned an empty string. Only that bech32 will print the error stack

Comment (iulianpascalau) [reply node/nodeTransactions.go:345]: Will prevent the log.Warn print

File: node/nodeTransactions_test.go (hunk 2 above)
Comment (iulianpascalau) [reply node/nodeTransactions_test.go:492]: done

**Review by bogdan-rosianu — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**